### PR TITLE
emqtt Truststore

### PR DESCRIPTION
**Pull request by lincmba**
Use PKCS12 which is an industry standard format instead of JKS keystore standard.

Generate truststore too, necessary for consume MQTT to authenticate

### CODE REVIEW

File: tasks/config.yml [hunk 1]
```diff
@@ -52,13 +52,30 @@
...
-  java_keystore:
-    name: certificates
-    certificate: "{{ nifi_emqtt_certificate_authority_cert }}"
-    private_key: "{{ nifi_emqtt_certificate_authority_key }}"
-    password: "{{ nifi_emqtt_secure_password }}"
-    dest: "{{ nifi_emqtt_kestore_file_dest }}"
+- name: Copy emqtt certificate and key
+  copy:
+    content: "{{ nifi_emqtt_certificate_authority_key }}\n{{ nifi_emqtt_certificate_authority_cert }}"
+    dest: "{{ nifi_emqtt_cert_dir }}/key_cert.pem.txt"
```
Comment (moshthepitt) [inline tasks/config.yml:58]: Do we need to keep this on the server after we are done?

Comment (lincmba) [reply tasks/config.yml:58]: Probably not. Let me get rid of them.

**Review by moshthepitt — CHANGES_REQUESTED** [review]
Lets add some documentation on certs and mqtt in the README in this PR

**Comment by moshthepitt** [comment]
Lets do these:

1. In this file: https://github.com/onaio/ansible-nifi/blob/mqtt-ssl-certs/defaults/main.yml could we separate the emqtt cert variables from the parameter context variables
2. Can we only run this task https://github.com/onaio/ansible-nifi/blob/mqtt-ssl-certs/tasks/config.yml#L41 when `nifi_create_mqtt_certs` is true?  I suppose we don't need the root cert if we are not going to use it - right?

**Comment by lincmba** [comment]
> 2. Can we only run this task https://github.com/onaio/ansible-nifi/blob/mqtt-ssl-certs/tasks/config.yml#L41 when nifi_create_mqtt_certs is true? I suppose we don't need the root cert if we are not going to use it - right?

This isn't introduced by this PR. Could be something necessary for NiFi elsewhere

**Review by moshthepitt — APPROVED** [review]

**Pull request merged**